mongodbatlas/data_source_mongodbatlas_federated_database_instance.go

```diff
@@ -0,0 +1,304 @@
+											Type:     schema.TypeList,
+											Computed: true,
+											Elem: &schema.Schema{
+												Type: schema.TypeString,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				"views": {
+					Type:     schema.TypeSet,
+					Computed: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"name": {
+								Type:     schema.TypeString,
+								Computed: true,
+							},
+							"source": {
+								Type:     schema.TypeString,
+								Computed: true,
+							},
+							"pipeline": {
+								Type:     schema.TypeString,
+								Computed: true,
+							},
+						},
+					},
+				},
+				"max_wildcard_collections": {
+					Type:     schema.TypeInt,
+					Computed: true,
+				},
+			},
+		},
+	}
+}
+
+func schemaFederatedDatabaseInstanceStoresDataSource() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"name": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"provider": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"region": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"bucket": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"cluster_name": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"project_id": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"prefix": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"delimiter": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"include_tags": {
+					Type:     schema.TypeBool,
+					Computed: true,
+				},
+				"additional_storage_classes": {
+					Type:     schema.TypeList,
+					Computed: true,
+					Elem:     &schema.Schema{Type: schema.TypeString},
+				},
+				"read_preference": {
+					Type:     schema.TypeList,
+					Computed: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"mode": {
+								Type:     schema.TypeString,
+								Computed: true,
+							},
+							"max_staleness_seconds": {
+								Type:     schema.TypeInt,
+								Computed: true,
+							},
+							"tags": {
+								Type:     schema.TypeList,
+								Computed: true,
+								Elem: &schema.Resource{
+									Schema: map[string]*schema.Schema{
+										"name": {
+											Type:     schema.TypeString,
+											Computed: true,
+										},
+										"value": {
+											Type:     schema.TypeString,
+											Computed: true,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceMongoDBAtlasFederatedDatabaseInstanceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*MongoDBClient).Atlas
+
+	projectID := d.Get("project_id").(string)
+	name := d.Get("name").(string)
```

Comment:
Can we rename this to `tenantName` as in the [API docs](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Data-Federation/operation/deleteFederatedDatabase)?

Reply:
I see we use name also in other resources instead of the field name in the API (for instance [resource_mongodbatlas_data_lake.go#L39](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/882524f7d4c3fec18302846127d29bfab56eba6f/mongodbatlas/resource_mongodbatlas_data_lake.go#L39))